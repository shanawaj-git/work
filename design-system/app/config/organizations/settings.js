import { Organization } from './organization';
import defaultSettings from './settings/default';
import mpcSettings from './settings/mpc';
import automotiveSettings from './settings/automotive';

const ORGANIZATIONS_TO_SETTINGS_MAP = {
  [Organization.DEFAULT]: defaultSettings,
  [Organization.MPC]: mpcSettings,
  [Organization.AUTOMOTIVE]: automotiveSettings,
};

const organizationSettings = Object.keys(Organization).reduce(
  (res, organization) => {
    const baseOrganizationSettings =
      ORGANIZATIONS_TO_SETTINGS_MAP[organization];

    res[organization] = {
      ...baseOrganizationSettings,
      isDefaultOrganization: organization === Organization.DEFAULT,
    };

    return res;
  },
  {},
);

const SETTINGS = organizationSettings;

export { SETTINGS };
