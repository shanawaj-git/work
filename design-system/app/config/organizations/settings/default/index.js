import { theme } from './theme';

const settings = {
  name: 'Main',
  // more settings to be added
};

export default { ...settings, theme };
