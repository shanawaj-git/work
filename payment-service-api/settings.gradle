rootProject.name = "payment-service-api"