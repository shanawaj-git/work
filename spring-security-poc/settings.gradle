rootProject.name = 'spring-security-poc'
