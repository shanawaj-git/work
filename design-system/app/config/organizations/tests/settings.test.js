import { SETTINGS } from '../settings';

describe('organization settings', () => {
  it('Should return organization settings if organization exists', () => {
    expect(SETTINGS.DEFAULT).toMatchSnapshot();
  });

  it("Should return undefined object if organization doesn't exist exists", () => {
    expect(SETTINGS.UNKNOWN).toMatchSnapshot();
  });
});
