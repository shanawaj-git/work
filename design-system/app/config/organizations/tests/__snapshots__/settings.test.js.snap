// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organization settings Should return organization settings if organization exists 1`] = `
Object {
  "isDefaultOrganization": true,
  "name": "Main",
  "theme": Object {
    "color": Object {
      "c100": "#E25F6D",
      "c200": "#FFFFFF",
      "c300": "#E8E8E6",
      "c400": "#9A9D9D",
      "c500": "#495151",
      "c600": "#d4d9d9",
      "p100": "#F6FAFB",
      "p200": "#FCF7F7",
      "p300": "#E25F6D",
      "p400": "#535858",
      "p500": "#020A0A",
    },
    "overrides": Object {
      "button": Object {
        "borderRadius": "32px",
      },
    },
  },
}
`;

exports[`organization settings Should return undefined object if organization doesn't exist exists 1`] = `undefined`;
